runners/apex/src/main/java/org/apache/beam/runners/apex/translation/operators/ApexParDoOperator.java

```diff
@@ -83,6 +105,11 @@
   @Bind(JavaSerializer.class)
   private final List<PCollectionView<?>> sideInputs;
 
+  private StateInternalsProxy<?> currentKeyStateInternals;
+  // TODO: fault tolerant timer internals
```

Comment:
The timer state is transient, if the operator gets restored to checkpointed state due to a failure, the timer state is lost. I'm going to add checkpointed timer state in the subsequent SDF PR.